File: src/main/java/com/wafflestudio/draft/model/response/RoomResponse.java
package com.wafflestudio.draft.model.response;

import com.wafflestudio.draft.model.Room;
import com.wafflestudio.draft.model.enums.RoomStatus;
import lombok.Data;

import java.time.LocalDateTime;


@Data
public class RoomResponse {
    private Long id;
    private RoomStatus roomStatus;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private String name;
    private LocalDateTime createdAt;
    private Long ownerId;
    private Long courtId;

    public RoomResponse(Room room) {
        this.id = room.getId();
        this.roomStatus = room.getStatus();
        this.startTime = room.getStartTime();
        this.endTime = room.getEndTime();
        this.name = room.getName();
        this.createdAt = room.getCreatedAt();
        this.ownerId = room.getOwner().getId();
        this.courtId = room.getCourt().getId();
    }
}

File: src/main/kotlin/com/wafflestudio/draft/dto/response/RoomResponse.kt
package com.wafflestudio.draft.dto.response

import com.wafflestudio.draft.model.Room
import com.wafflestudio.draft.model.enums.RoomStatus
import java.time.LocalDateTime

data class RoomResponse(
        var id: Long? = null,
        var roomStatus: RoomStatus? = null,
        var startTime: LocalDateTime? = null,
        var endTime: LocalDateTime? = null,
        var name: String? = null,
        var createdAt: LocalDateTime? = null,
        var ownerId: Long? = null,
        var courtId: Long? = null,
        var participants: List<UserInformationResponse>? = null
) {
    constructor(room: Room): this() {
        this.id = room.id
        this.roomStatus = room.status
        this.startTime = room.startTime
        this.endTime = room.endTime
        this.name = room.name
        this.createdAt = room.createdAt
        this.ownerId = room.owner!!.id
        this.courtId = room.court!!.id
        val userResponses: MutableList<UserInformationResponse> = mutableListOf()
        room.participants.map { userResponses.add(UserInformationResponse(it.user)) }
        this.participants = userResponses
    }
}
